experimental/iterators/CMakeLists.txt

```diff
@@ -23,6 +30,8 @@ target_include_directories(iterators PUBLIC include)
 include(CTest)
 enable_testing()
 
+# TODO: move these to the tests director.
+# TODO: split tests into C++ and MLIR
```

Comment:
Will do. I saw that LLVM uses googletests for some things (in the unittest directories). I guess I can adopt that pattern.